packages/core/test/actor.test.ts

```diff
@@ -1105,4 +1106,75 @@ describe('actors', () => {
       })
       .start();
   });
+
+  // https://github.com/statelyai/xstate/issues/2507
+  it('should not attempt to clean up children before stopping', () => {
+    const grandChildMachine = createMachine({
+      initial: 'idle',
+      states: {
+        idle: {
+          always: [
+            {
+              target: 'stopped'
+            }
+          ]
+        },
+        stopped: {
+          type: 'final'
+        }
+      }
+    });
+    const childMachine = createMachine<any>(
+      {
+        initial: 'idle',
+        context: {
+          child: null
+        },
+        states: {
+          idle: {
+            entry: ['setup']
+          }
+        }
+      },
+      {
+        actions: {
+          setup: assign({
+            child: () => spawn(grandChildMachine)
+          })
+        }
+      }
+    );
+
+    const parentMachine = createMachine(
+      {
+        initial: 'idle',
+        context: {
+          child: null
+        },
+        states: {
+          idle: {
+            on: {
+              SETUP: {
+                actions: ['setup']
+              }
+            }
+          }
+        }
+      },
+      {
+        actions: {
+          setup: assign({
+            child: () => spawn(childMachine)
+          })
+        }
+      }
+    );
+
+    const service = interpret(parentMachine);
+    service.start();
+
+    expect(() => {
+      service.send('SETUP');
+    }).not.toThrow();
+  });
```

Comment:
A potentially simpler test case can be written for to validate this behavior:
```suggestion
  // https://github.com/statelyai/xstate/issues/2507
  it('should not attempt to clean up children before stopping', () => {
    const childMachine = createMachine({
      initial: 'idle',
      states: {
        idle: {
          always: [
            {
              target: 'stopped'
            }
          ]
        },
        stopped: {
          type: 'final'
        }
      }
    });

    const parentMachine = createMachine(
      {
        context: {
          child: null
        },
        entry: 'setup'
      },
      {
        actions: {
          setup: assign({
            child: () => spawn(childMachine)
          })
        }
      }
    );

    const service = interpret(parentMachine);

    expect(() => {
      service.start();
    }).not.toThrow();
  });
```

Reply:
It roughly tests for the same thing with just a parent and a child instead of using a 3 layer actor hierarchy